### DISCUSSION TITLE
Trouble installing plugins hosted on gitlab

### DISCUSSION THREAD

**Discussion by samEngineering** [wbthomason/packer.nvim#990] [General]
I'm not sure what's going on, but I have made a simple plugin and host it on gitlab. I get "failed to install ... " when runningn PackerUpdate with the repo listed in my packages.lua file. To test this, I put the same plugin on github and it worked with no problems. Does anyone know why that is happening? 

I'm on Arch Linux and most recent neovim.

**Comment by EdenEast** [comment]
The [default_url_format](https://github.com/wbthomason/packer.nvim/blob/master/lua/packer.lua#L48) is github. Packer [checks](https://github.com/wbthomason/packer.nvim/blob/00ec5adef58c5ff9a07f11f45903b9dbbaa1b422/lua/packer/plugin_utils.lua#L26-L31) if you gave it a url. If you dont give a url it defaults to the `default_url_format`. To use for gitlab plugin you would just have to specify the complete url.


```lua
require("packer").startup(function(use)
 use("https://gitlab.com/user/repo")
end)
```

**Reply by nyngwang** [reply]
Thanks for your help!

**Comment by samEngineering** [comment]
Ah! Thanks friend.
